compiler/noirc_evaluator/src/ssa/ir/printer.rs

```diff
@@ -173,8 +173,8 @@ fn display_instruction_inner(
         }
         Instruction::Allocate => writeln!(f, "allocate"),
         Instruction::Load { address } => writeln!(f, "load {}", show(*address)),
-        Instruction::Store { address, value } => {
-            writeln!(f, "store {} at {}", show(*value), show(*address))
+        Instruction::Store { address, value, from_rc } => {
+            writeln!(f, "store {} at {}, from_rc {}", show(*value), show(*address), *from_rc)
```

Comment:
I'm not a big fan of including `from_rc` on every store instruction for a mem2reg-specific issue.

Can we add tracking to mem2reg specifically instead? E.g. if we see a load -> dec-rc -> store we mark that we can't remove the store?

Reply:
I'll look at switching to that

Reply:
I switched to tracking the current rc reload per instruction with an `Option<(ValueId, bool)>`.

Reply:
Ah something looks to be failing. I had it passing before but I guess I made a bad change while cleaning up.

Reply:
I had moved when I was calling my method to track the rc reload state, but this led to inadvertently calling the method on the wrong instruction id. This is now fixed and the PR is ready for review again.

Reply:
Looks to still be failing on the debugger actually as the debugger inserts foreign calls which are breaking up the expected block of this form:
```
    v72 = load v57
    inc_rc v72
    v73 = load v57
    inc_rc v73
    store v73 at v57
    store v72 at v60
```
In the debugger we see the following after mem2reg:
```
    v80 = load v65
    inc_rc v80
    call v44(Field 1, v80)
    inc_rc v80
    store v80 at v67
```